src/store/setToolMode.test.js

```diff
@@ -0,0 +1,89 @@
+import getToolForElement from './getToolForElement.js';
+import { setToolEnabledForElement } from './setToolMode.js';
+
+jest.mock('./getToolForElement.js');
+
+describe('setToolModeForElement', () => {
+  it('sets the tool.mode to enabled', () => {
+    const mergeOptions = jest.fn();
+    const testTool = {
+      mode: undefined,
+      options: { mouseButtonMask: [] },
+      supportedInteractionTypes: ['Mouse'],
+      mergeOptions,
+    };
+
+    getToolForElement.mockImplementationOnce(() => testTool);
```

Comment:
My JS Unit testing "best practice" knowledge is hazy. It might be better to setup the app state for this scenario instead of mocking implementation behavior. For example:

> We expect X to happen when the application's state is Y

Will hopefully be less brittle than:

> We expect X to happen when implementation detail returns Z

I'm very rusty here, but it may be worth discussing.

Reply:
Maybe the problem here is that the functionality that I want to test is hidden inside something bigger. So because I can't test the unit I have to test something bigger...

Reply:
I'm going to push another commit to this branch which might shine some light on this and also expose another potential bug.

Reply:
Also, the more I look at this, the more I wonder why merging these arrays is the right thing to do here.

Reply:
See PR with more tests here #903